scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/termination/handlers/TerminateIfTaskHandler.java

```diff
@@ -0,0 +1,190 @@
+		// notify frontend that tasks were modified
+		if (frontend != null) {
+			frontend.jobStateUpdated(internalJob.getOwner(),
+					new NotificationData<>(SchedulerEvent.TASK_SKIPPED, internalJob.getJobInfo()));
+		}
+		((JobInfoImpl) internalJob.getJobInfo()).clearTasksChanges();
+
+		// no jump is performed ; now that the tasks have been plugged
+		// the flow can continue its normal operation
+		internalJob.getJobDescriptor().terminate(taskId);
+
+		return true;
+	}
+
+	private InternalTask[] searchIfElseJoinTasks(FlowAction action, InternalTask initiator) {
+		InternalTask targetIf = null;
+		InternalTask targetElse = null;
+		InternalTask targetJoin = null;
+
+		// search for the targets as perfect matches of the unique name
+		for (InternalTask it : internalJob.getIHMTasks().values()) {
+
+			// target is finished : probably looped
+			if (it.getStatus().equals(TaskStatus.FINISHED) || it.getStatus().equals(TaskStatus.SKIPPED)) {
+				continue;
+			}
+			if (action.getTarget().equals(it.getName())) {
+				if (it.getIfBranch().equals(initiator)) {
+					targetIf = it;
+				}
+			} else if (action.getTargetElse().equals(it.getName())) {
+				if (it.getIfBranch().equals(initiator)) {
+					targetElse = it;
+				}
+			} else if (action.getTargetContinuation().equals(it.getName())) {
+				InternalTask up = internalJob.findTaskUp(initiator.getName(), it);
+				if (up != null && up.equals(initiator)) {
+					targetJoin = it;
+				}
+			}
+
+		}
+
+		boolean searchIf = (targetIf == null);
+		boolean searchElse = (targetElse == null);
+		boolean searchJoin = (targetJoin == null);
+
+		// search of a runnable perfect match for the targets failed;
+		// the natural target was iterated, need to find the next iteration
+		// which is the the one with the same dup index and base name,
+		// but the highest iteration index
+		for (InternalTask it : internalJob.getIHMTasks().values()) {
+
+			// does not share the same dup index : cannot be the same scope
+			if (it.getReplicationIndex() != initiator.getReplicationIndex()) {
+				continue;
+			}
+
+			if (it.getStatus().equals(TaskStatus.FINISHED) || it.getStatus().equals(TaskStatus.SKIPPED)) {
+				continue;
+			}
+
+			String name = InternalTask.getInitialName(it.getName());
+
+			if (searchIf && InternalTask.getInitialName(action.getTarget()).equals(name)) {
+				if (targetIf == null || targetIf.getIterationIndex() < it.getIterationIndex()) {
+					targetIf = it;
+				}
+			} else if (searchElse && InternalTask.getInitialName(action.getTargetElse()).equals(name)) {
+				if (targetElse == null || targetElse.getIterationIndex() < it.getIterationIndex()) {
+					targetElse = it;
+				}
+			} else if (searchJoin && InternalTask.getInitialName(action.getTargetContinuation()).equals(name)) {
+				if (targetJoin == null || targetJoin.getIterationIndex() < it.getIterationIndex()) {
+					targetJoin = it;
+				}
+			}
+		}
+
+		InternalTask[] result = { targetIf, targetElse, targetJoin };
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)